sdk/core/core/inc/az_span.h

```diff
@@ -32,25 +37,61 @@ typedef struct {
 
 AZ_STATIC_ASSERT(CHAR_BIT == 8);
 
-// Returns a byte in `index` position.
-inline uint8_t az_const_span_get(az_const_span const span, size_t const index) {
-  AZ_ASSERT(index < span.size);
+/**
+ * Returns a byte in `index` position.
+ * Returns `AZ_OPTION_BYTE_NONE` if the `index` is out of the span range.
+ */
+inline az_option_byte az_const_span_get(az_const_span const span, size_t const index) {
+  if (span.size <= index) {
+    return AZ_OPTION_BYTE_NONE;
+  }
   return span.begin[index];
 }
 
-// Returns a sub span of the given span.
-inline az_const_span az_const_sub_span(az_const_span const span, size_t const from, size_t const to) {
-  AZ_ASSERT(from <= to);
-  AZ_ASSERT(to <= span.size);
-  return (az_const_span){ .begin = span.begin + from, .size = to - from };
+/**
+ * \brief returns a span with the left \n bytes of the given \span.
```

Comment:
At some point we need to go full doxygen for all comments, either with your pr, or a separate PR.

Reply:
I believe we also specify to use the @command doxygen syntax in the style guide

Reply:
#41